app/src/main/kotlin/com/mapzen/erasermap/controller/MainActivity.kt

```diff
@@ -652,6 +622,9 @@ class MainActivity : AppCompatActivity(), MainViewController,
      * to show or hide debug settings in settings fragment
      */
     override fun toggleShowDebugSettings() {
+        if (!AndroidAppSettings.SHOW_DEBUG_SETTINGS_QUERY.equals(searchController.searchView?.query.toString())) {
```

Comment:
In a recent pull request this logic was moved into the presenter with the idea being that `toggleShowDebugSettings()` should not be called unless this check had already been made. Is that not the case anymore?

https://github.com/mapzen/eraser-map/pull/787